pkg/scheduler/framework/plugins/tainttoleration/filtering_test.go

```diff
@@ -0,0 +1,410 @@
+package tainttoleration
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clusterv1beta1 "go.goms.io/fleet/apis/cluster/v1beta1"
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"go.goms.io/fleet/pkg/scheduler/framework"
+)
+
+var (
+	cmpStatusOptions = cmp.Options{
+		cmpopts.IgnoreFields(framework.Status{}, "err"),
+		cmp.AllowUnexported(framework.Status{}),
+	}
+)
+
+func TestFilter(t *testing.T) {
+	p := New()
+	tests := []struct {
+		name       string
+		cluster    *clusterv1beta1.MemberCluster
+		placement  *placementv1beta1.ClusterSchedulingPolicySnapshot
+		wantStatus *framework.Status
+	}{
+		{
+			name: "empty taints",
+			cluster: &clusterv1beta1.MemberCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-mc",
+				},
+				Spec: clusterv1beta1.MemberClusterSpec{},
+			},
+			placement: &placementv1beta1.ClusterSchedulingPolicySnapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "csp-1",
+				},
+				Spec: placementv1beta1.SchedulingPolicySnapshotSpec{
+					Policy: &placementv1beta1.PlacementPolicy{
+						PlacementType: placementv1beta1.PickAllPlacementType,
+					},
+				},
+			},
+			wantStatus: nil,
+		},
+		{
+			name: "empty tolerations",
+			cluster: &clusterv1beta1.MemberCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-mc",
+				},
+				Spec: clusterv1beta1.MemberClusterSpec{
+					Taints: []clusterv1beta1.Taint{
+						{
+							Key:    "key1",
+							Value:  "value1",
+							Effect: corev1.TaintEffectNoSchedule,
+						},
+					},
+				},
+			},
+			placement: &placementv1beta1.ClusterSchedulingPolicySnapshot{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "csp-1",
+				},
+				Spec: placementv1beta1.SchedulingPolicySnapshotSpec{
+					Policy: &placementv1beta1.PlacementPolicy{
+						PlacementType: placementv1beta1.PickAllPlacementType,
+					},
+				},
+			},
+			wantStatus: framework.NewNonErrorStatus(framework.ClusterUnschedulable, p.Name(), fmt.Sprintf(reasonFmt, &clusterv1beta1.Taint{Key: "key1", Value: "value1", Effect: corev1.TaintEffectNoSchedule})),
+		},
+		{
+			name: "taints can be tolerated based on key, value & effect with Equal operator, every taint has matching toleration - nil status",
```

Comment:
add some (positive and negative) tests when taint key is present + empty value + operation is equal.